### ISSUE TITLE
Specific part of Gemfile breaks `jekyll serve`

### ISSUE THREAD

**Issue by cmegown** [jekyll/jekyll#4818]
Hi everyone,

I'm fairly certain this is a Jekyll issue, or perhaps an annoying Win8 issue. I added a Gemfile via the instructions [here](https://help.github.com/articles/setting-up-your-github-pages-site-locally-with-jekyll/#step-2-install-jekyll-using-bundler):

```
source 'https://rubygems.org'

gem 'github-pages', group: :jekyll_plugins
gem 'wdm', '>= 0.1.0' if Gem.win_platform?
```

I ran `bundle exec jekyll serve` and received this error:

```
Configuration file: F:/GitHub/cmegown.github.io/_config.yml
       Deprecation: Auto-regeneration can no longer be set from your configuration file(s). Use the
--[no-]watch/-w command-line option instead.
            Source: F:/GitHub/cmegown.github.io
       Destination: F:/GitHub/cmegown.github.io/_site
 Incremental build: disabled. Enable with --incremental
      Generating...
jekyll 3.0.3 | Error:  No such file or directory - git remote --verbose
```

There's nothing more than a name declared in my `_config.yml` file at this point so I know there's no problem with that. Next I thought that maybe it was a Bundler issue so I ran `jekyll serve`, which gave me the same error.

Long story short I found that removing a specific part of the Gemfile fixes the issue, and I'd appreciate help understanding why as well as how to fix it.

Here's the updated Gemfile:

```
source 'https://rubygems.org'

gem 'github-pages'
gem 'wdm', '>= 0.1.0' if Gem.win_platform?
```

So removing the `, group: :jekyll_plugins` resolves the issue and allows me to build...but why?

**Comment by envygeeks** [comment]
Unfortunately this is not a Jekyll (Core) issue.  When you add it to the `:jekyll_plugins` group then you get it auto-required (all of the gems that pages currently supports... plugins too.) One of those gems (or perhaps bundler itself -- though doubtful) gets required and throws an error.

There was a recent issue with git involving the I think it was the github-meta gem? That might be causing this error, perhaps @parkr, @benbalter or @pathawks can let you know if this is caused by that.

You, yourself confirmed it probably isn't a Core issue by removing it from the `:jekyll_plugins` group which removed all of the possible plugins that `github-pages` would use (unless you explicitly added them to your `gems` list in `_config.yml` and the error went away.

**Comment by cmegown** [comment]
Thanks @envygeeks for the explanation. This is the first time I've used Bundler and I suppose I'm still trying to understand exactly what's going on. I'm not even sure what the `:jekyll plugins` does, but was under the impression is enforced versions or dependencies or something.

I do have the jekyll-feed and jekyll-sitemap gems listed in my `config.yml` file now, but they don't seem to have an effect on whether or not `:jekyll_plugins` is in my Gemfile or not.

**Comment by envygeeks** [comment]
@cmegown The `:jekyll_plugins` group is a special group that we auto-require on your behalf.  It's way for you not to have to manage your plugins inside of the `gems` inside of your `_config.yml`.  So if you add something to the `:jekyll_plugins` group and we will require it for you through Bundler (AKA: `Bundle.require(:jekyll_plugins)`)

When you remove it from that group, all plugins become manual.  Meaning you must do:

```yml
gems:
  - jekyll-feed
```

When you removed it from that group and it worked suddenly, diagnostically that told me that it's probably not directly a Core (Jekyll) issue but probably caused by a plugin somewhere.

**Comment by cmegown** [comment]
Excellent response, thanks for clarifying that for me! Now that I understand what's going on I can continue on my merry way, but I'll leave it up to whoever to decide if this issue should remain open as it does still seem like there's a bug somewhere along the way.

**Comment by parkr** [comment]
@envygeeks Yeah, there was recently an issue with this. But it wasn't this, it was about an empty response to `git remote --verbose`: https://github.com/jekyll/github-metadata/pull/54

I have no clue how you'd get what you're getting, @cmegown, as we use backticks (http://ruby-doc.org/core-2.3.0/Kernel.html#method-i-60) to execute `git remote --verbose`. What version of `github-pages` are you running (`bundle show github-pages`), and do you have any plugins which override that Kernel backtick method?

**Comment by parkr** [comment]
This is an issue with the plugin, so I'm going to close this.
